lib/deePool.src.js

```diff
@@ -29,25 +29,25 @@
 		// ******************************
 
 		function use() {
-      if (freePoolEnd == -1) {
-        if (freePool.length == 0) {
-          grow(5);
-        } else {
-          grow();
-        }
-      }
-      return freePool[freePool.length - (freePoolEnd--)];
+			if (freePoolEnd == -1) {
+				if (freePool.length == 0) {
+					grow(5);
+				} else {
+					grow();
+				}
+			}
+			return freePool[freePool.length - 1 - (freePoolEnd--)];
```

Comment:
do we need to subtract `freePoolEnd` from the length, or could it just be the pointer to the location itself... like `return freePool[ freePoolEnd-- ];`